internal/impl/gcp/processor_vertex_ai.go

```diff
@@ -0,0 +1,316 @@
+	vaiFieldPrompt           = "prompt"
+	vaiFieldSystemPrompt     = "system_prompt"
+	vaiFieldTemp             = "temperature"
+	vaiFieldTopP             = "top_p"
+	vaiFieldTopK             = "top_k"
+	vaiFieldMaxTokens        = "max_tokens"
+	vaiFieldStop             = "stop"
+	vaiFieldPresencePenalty  = "presence_penalty"
+	vaiFieldFrequencyPenalty = "frequency_penalty"
+	vaiFieldResponseFormat   = "response_format"
+)
+
+func init() {
+	err := service.RegisterProcessor(
+		"gcp_vertex_ai_chat",
+		newVertexAIProcessorConfig(),
+		newVertexAIProcessor,
+	)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func newVertexAIProcessorConfig() *service.ConfigSpec {
+	return service.NewConfigSpec().
+		Categories("AI").
+		Summary("Generates responses to messages in a chat conversation, using the Vertex AI API.").
+		Description(`This processor sends prompts to your chosen large language model (LLM) and generates text from the responses, using the Vertex AI API.
+
+For more information, see the https://cloud.google.com/vertex-ai/docs[Vertex AI documentation^].`).
+		Version("4.33.0").
+		Fields(
+			service.NewStringField(vaiFieldProject).
+				Description("GCP project ID to use"),
+			service.NewStringField(vaiFieldCredentialsJSON).
+				Description("An optional field to set google Service Account Credentials json.").
+				Secret().
+				Optional(),
+			service.NewStringField(vaiFieldLocation).
+				Description("The location of the model if using a fined tune model. For base models this can be omitted").
+				Optional().
+				Examples("us-central1"),
+			service.NewStringField(vaiFieldModel).
+				Description("The name of the LLM to use. For a full list of models, see the https://console.cloud.google.com/vertex-ai/model-garden[Vertex AI Model Garden].").
+				Examples("gemini-1.5-pro-001", "gemini-1.5-flash-001"),
+			service.NewInterpolatedStringField(vaiFieldPrompt).
+				Description("The prompt you want to generate a response for. By default, the processor submits the entire payload as a string.").
+				Optional(),
+			service.NewInterpolatedStringField(vaiFieldSystemPrompt).
+				Description("The system prompt to submit to the Vertex AI LLM.").
+				Advanced().
+				Optional(),
+			service.NewFloatField(vaiFieldTemp).
+				Description("Controls the randomness of predications.").
+				Optional(),
+			service.NewIntField(vaiFieldMaxTokens).
+				Description("The maximum number of output tokens to generate per message.").
+				Optional(),
+			service.NewStringEnumField(vaiFieldResponseFormat, "text", "json").
+				Description("The response format of generated type, the model must also be prompted to output the appropriate response type.").
+				Default("text"),
+			service.NewFloatField(vaiFieldTopP).
+				Advanced().
+				Description("If specified, nucleus sampling will be used.").
+				Optional(),
+			service.NewIntField(vaiFieldTopK).
+				Advanced().
+				Description("If specified top-k sampling will be used.").
+				Optional(),
```

Comment:
Would be nice to add linting for these like in the other PR

Reply:
Done - some of these values differ based on the model used, but I've just used the biggest range for each.